inbm/dispatcher-agent/tests/unit/sota/test_downloader.py

```diff
@@ -110,3 +114,39 @@ def _build_mock_repo(num_files=0):
             for i in range(0, num_files):
                 mem_repo.add("test" + str(i + 1) + ".rpm", b"0123456789")
         return mem_repo
+
+    @patch('dispatcher.sota.downloader.oras_download')
+    def test_tiberos_download_successful(self, mock_download) -> None:
+        self.release_date = self.username = None
+        password = "mock_password"
+        mock_url = canonicalize_uri("https://registry-rs.internal.ledgepark.intel.com/one-intel-edge/tiberos:latest")
+
+        assert TestDownloader.sota_instance
+        TestDownloader.sota_instance.factory = SotaOsFactory(
+            MockDispatcherBroker.build_mock_dispatcher_broker(), None, []).get_os('TiberOS')
+        factory = TestDownloader.sota_instance.factory
+        assert factory
+        installer = factory.create_downloader()
+        assert isinstance(installer, TiberOSDownloader)
+
+        if not os.path.exists(CACHE):
+            os.makedirs(CACHE)
+
+        file_path = os.path.join(CACHE, 'test')
+        with open(file_path, 'w') as file:
```

Comment:
unit tests should not interact with network/filesystem (but integration tests can)

Reply:
(the reasons for this: 1. can slow them down; 2. can make them break if there's no file access on dev system; 3. can cause test-order dependency)

Reply:
Updated. Use tempfile in unit test.